hyperspy/signal.py

```diff
@@ -3867,6 +3867,157 @@ def add_gaussian_noise(self, std):
+            navigation_axes = nat_axes[:-signal_axes]
+            signal_axes = nat_axes[-signal_axes:]
+        elif iterable_not_string(signal_axes):
+            if navigation_axes is None:
+                signal_axes = tuple(am[ax] for ax in signal_axes)
+                navigation_axes = tuple(ax for ax in nat_axes if ax not in
+                                        signal_axes)
+            elif iterable_not_string(navigation_axes):
+                # want to keep the order
+                signal_axes = tuple(am[ax] for ax in signal_axes)
+                navigation_axes = tuple(am[ax] for ax in navigation_axes)
+                if len(set(signal_axes).intersection(navigation_axes)):
+                    raise ValueError("axis either signal or navigation, not"
+                                     " both")
+                if len(am._axes) != (len(signal_axes) + len(navigation_axes)):
+                    raise ValueError("Not enough axes speficied")
+            else:
+                raise ValueError("iterable + int given. two iterables or one"
+                                 " int allowed")
+        elif signal_axes is None:
+            if isinstance(navigation_axes, int):
+                signal_axes = nat_axes[navigation_axes:]
+                navigation_axes = nat_axes[:navigation_axes]
+            elif iterable_not_string(navigation_axes):
+                navigation_axes = tuple(am[ax] for ax in navigation_axes)
+                signal_axes = tuple(ax for ax in nat_axes if ax not in
+                                    navigation_axes)
+            elif navigation_axes is None:
+                signal_axes = am.navigation_axes
+                navigation_axes = am.signal_axes
+            else:
+                raise ValueError("The passed options are not valid")
+        else:
+            raise ValueError("The passed options are not valid")
+        # get data view
+        array_order = tuple(
+            ax.index_in_array for ax in reversed(navigation_axes))
+        array_order += tuple(ax.index_in_array for ax in reversed(signal_axes))
+        newdata = self.data.transpose(array_order)
+
+        res = self._deepcopy_with_new_data(newdata)
+
+        # reconfigure the axes of the axesmanager:
+        axes_order = tuple(
+            ax.index_in_axes_manager for ax in reversed(navigation_axes))
+        axes_order += tuple(ax.index_in_axes_manager for ax in reversed(signal_axes))
+
+        ram = res.axes_manager
+        ram._update_trait_handlers(remove=True)
+        ram._axes = [ram._axes[i] for i in axes_order]
+        for i, ax in enumerate(ram._axes):
+            if i < len(navigation_axes):
+                ax.slice = None
+                ax.navigate = True
+            else:
+                ax.slice = slice(None)
+                ax.navigate = False
+        ram._update_attributes()
+        ram._update_trait_handlers(remove=False)
+        res.get_dimensions_from_data()
```

Comment:
I definitely do (both reorder and need).

Though hopefully I'm doing the axes reordering properly...